File: Buttononclickfunc/app/src/main/java/com/example/muskangoyal/buttononclickfunc/MainActivity.java
package com.example.muskangoyal.buttononclickfunc;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity implements View.OnClickListener{

    public static final String TAG ="Button";
    Button button2,button3;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        button2 = findViewById(R.id.button2);
        button3=findViewById(R.id.button3);

        button2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Log.d(TAG,"second button clicked");
            }
        });
        button3.setOnClickListener(this);
    }

    public void onButtonClick(View view){
        Log.d(TAG,"button is clicked");
    }

    @Override
    public void onClick(View view) {
      Log.d(TAG,"third button clicked");
    }
}

File: TodoListDatabase/app/src/main/java/com/example/muskangoyal/todolistdatabase/MainActivity.kt
package com.example.muskangoyal.todolistdatabase

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.ArrayAdapter
import com.example.muskangoyal.todolistdatabase.db.MyDbHelper
import com.example.muskangoyal.todolistdatabase.db.TodoTable
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    val todos = ArrayList<Todo>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val todoAdapter = ArrayAdapter<Todo>(
                this,
                android.R.layout.simple_list_item_1,
                android.R.id.text1,
                todos
        )
        val db = MyDbHelper(this).writableDatabase

        fun refreshTodoList () {
            val todoList = TodoTable.getAllTodos(db)
            todos.clear()
            todos.addAll(todoList)
            todoAdapter.notifyDataSetChanged()
        }
        lvtodo.adapter = todoAdapter
        refreshTodoList()

        btnadd.setOnClickListener {
            val newTodo = Todo(
                    et.text.toString(),
                    false
            )
            TodoTable.insertTodo(db, newTodo)
            refreshTodoList()
        }
    }
}
